### PR TITLE
refactor: update d2 & d2-ui and use app-runtime-adapter-d2

### PR DESCRIPTION
**Pull request by mediremi**
Closes [TECH-1095](https://dhis2.atlassian.net/browse/TECH-1095)

Changes:

- Update `d2`
- Replace `d2-ui` with `@dhis2/d2-ui-xxx` packages
- Initialise `d2` using `@dhis2/app-runtime-adapter-d2`
- Fixed tests

### CODE REVIEW

File: src/app.js [hunk 1]
```diff
@@ -75,57 +49,27 @@ function getSystemSettings(d2) {
...
-    }, [])
-
-    if (loading) {
+    const { d2, d2Error } = useD2({
+        i18nRoot: './i18n',
+        onInitialized: async d2 => {
+            await addCustomModels(d2)
+            await getSystemSettings(d2)
+        }
+    })
```
Comment (mediremi) [inline src/app.js:59]: `getUserSettings`, `configI18n` and `init` are handled by `useD2`

File: src/forms/form-fields/geometry-field.js [hunk 2]
```diff
@@ -1,9 +1,11 @@
 import { Component } from 'react';
 import PropTypes from 'prop-types'
 import TextField from '../../forms/form-fields/text-field';
-import { isNumber } from 'd2-ui/lib/forms/Validators';
+import Validators from '@dhis2/d2-ui-forms/Validators';
 import { getOr } from 'lodash/fp';
 
+const { isNumber } = Validators
```
Comment (mediremi) [inline src/forms/form-fields/geometry-field.js:7]: Validators are exported with

```js
export default {
  isNumber,
  isUrl,
  // ....
}
```

so doing e.g. `import { isNumber } from '@dhis2/d2-ui-forms/Validators'` does not work

File: package.json [hunk 3]
```diff
@@ -90,14 +90,21 @@
     "@dhis2/app-runtime-adapter-d2": "^1.1.0",
     "@dhis2/cli-app-scripts": "^9.0.0",
     "@dhis2/d2-i18n": "^1.1.0",
-    "@dhis2/d2-ui-header-bar": "^1.1.4",
-    "@dhis2/d2-ui-sharing-dialog": "^1.0.12",
+    "@dhis2/d2-ui-core": "^7.3.4",
```
Comment (mediremi) [inline package.json:93]: `d2-ui` has been split up into and replaced by `@dhis2/d2-ui-xxx` packages

**Comment by mediremi** [comment]
Temporarily converting to draft since I need to update this branch after merging https://github.com/dhis2/maintenance-app/pull/2171

**Review by Mohammer5 — APPROVED** [review]
Seems to work :muscle:

**Pull request merged**